README.md

```diff
@@ -1,97 +1,228 @@
+  usually `your-client-application.com`.
+  Through this redirection, the toctoctoc server returns a GitHub
+  `access_token` to the client application.
+- From there, the client application can store the `access_token` (for instance,
+    in `localStorage`) and makes direct calls to `github.com`.
+
+You can also set it up for GitLab if you need it.
+
+## How to install toctoctoc
+
+### 1 - Prerequisites
+
+- You need to install [Node.js](https://nodejs.org/en/download/).
+- A URL to access your toctoctoc server. It can be `http://localhost:[port]` on
+  a development environment or a domain that points to your toctoctoc server.
+
+### 2 - Create an OAuth application
+
+Create an OAuth application for the forges you want to handle with the server:
+- A [GitHub OAuth app](https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/):
+  for the `Authorization callback URL`, use the endpoint provided by your
+  toctoctoc server: `[your-toctoctoc-server-URL]/github-callback`.
+- And/or a [GitLab OAuth app](https://docs.gitlab.com/ee/integration/oauth_provider.html):
+  for the `Callback URL`, use the endpoint provided by your toctoctoc server:
+  `[your-toctoctoc-server-URL]/gitlab-callback`.
+
+### 3 - Install the server
+
+Clone the repository
+```sh
+git clone git@github.com:Scribouilli/toctoctoc.git
+```
 
-After having logged in with github, this server forward everything useful (secret token) to the client-side. From there, the client-side code communicates directly with github without intermediaries (because github has CORS headers open and allows for these sorts of interactions)
+Install dependencies
+```sh
+npm install
+```
 
-The typical workflow goes like this:
-- go to `useful-service.com`
-    - this website contains a "login with github" button leading to `https://github.com/login/oauth/authorize?client_id=${client_id}&scope=${scope}&redirect_uri=${redirect_uri}`
-        - each different `useful-service.com` can customize the `scope` part of this link to ask only for the needed rights
-        - the `redirect_uri` must contain a `destination` parameter which is an absolute url to an allowed domain (usually back to `useful-service.com`)
-- on github.com, the user shares rights with toctoctoc
-- the toctoctoc server hands the keys back to the `redirect_uri`
-    - `useful-service.com` probably stores the keys in `localStorage` or equivalent
-- from there, the client-side of `useful-service.com` makes direct calls to `github.com`
+### 4 - Setup the environment variables
 
+You need to fill the client id and client secret of at least one forge service.
 
-## Benefits of this approach
+- `GITHUB_OAUTH_APP_CLIENT_ID`: GitHub OAuth application client id.
+- `GITHUB_OAUTH_APP_CLIENT_SECRET`: GitHub OAuth application client secret.
+- `GITLAB_OAUTH_APP_CLIENT_ID`: GitLab OAuth application id.
+- `GITLAB_OAUTH_APP_CLIENT_SECRET`: GitLab OAuth application secret.
```

Comment:
I feel this will be limiting if we want a toctoctoc which supports several gitlab instances (like gitlab.org & git.scribouilli.org & framagit.org for instance). I think we'll need a pair of env variables for each instance... and this doesn't grow well...

I'm not sure yet what a cleaner solution would be

And also, I'm fine with limiting the scope of this PR to a generic gitlab support and addressing multi-gitlab later